text/0006-equivocation-catastrophes.md

```diff
@@ -0,0 +1,114 @@
+
+As the last step, agree on the initial set of validators for the restarted network. The equivocators should be slashed, but there may be reasons to also exclude other validators, or even to start with a set of validators that has nothing in common with the validators from the stopped network. Once agreement has been reached, apply necessary changes to the pre-initial state: modify the validator stakes and change the entries in the auction contract. After this is applied, the resulting state will be the state used for the restart.
+
+After the users collectively decide upon the new initial state and the new initial set of validators, the network can be restarted.
+
+Note that tools would have to be provided that would facilitate the manual construction of a specific state. This could be realized in the form of node software constructing the state based on the provided hash of a block, list of block or deploy hashes defining deploys to be executed on top of that block, and possibly some additional upgrade code.
+
+### Restarting the network
+
+The process of restarting the network would be very similar to the process of starting it. In its essence, it is just like starting the network from scratch, but with a few differences:
+
+- There is a different set of genesis validators - the new validators chosen by the community.
+- There is a different initial global state: it is not empty anymore, but it is the initial state prepared before.
+- There is a different starting block height - we would not start from 0, but from the last safe block.
+- Other details like the protocol version can differ as well, but they are not of the main concern for the restart process itself.
+
+The agreed upon set of validators and the initial state would be included in the chainspec. The "restart-chainspec" (consisting of a new `chainspec.toml` along with any dependencies like `accounts.csv`) could be distributed among the validators the same way the initial chainspec for the network was. The files containing the new initial state could be distributed alongside the chainspec, or the nodes could download it from each other after they are started (which, of course, requires at least one node to possess the full copy of the state).
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+Here we will just summarize the exact steps to be performed when an equivocation catastrophe happens.
+
+1. An equivocation catastrophe is detected. Once a single validator detects it, we can expect all the validators to detect it eventually as well (even if just due to them receiving the evidence from the validator that detected it first).
+2. Consensus gets suspended, though validators continue to gossip. The users (validator nodes' owners) receive notice about the catastrophe and start the process of social consensus.
```

Comment:
It's worth noting that validators may detect equivocation catastrophe at different moments so their final protocol states may differ – some of them may even finalize few more units after others have detected the catastrophe already.

Reply:
> some of them may even finalize few more units after others have detected the catastrophe already

These would be the blocks whose finalization would be reverted by the equivocations, wouldn't they?